cps/kni.c

```diff
@@ -979,6 +982,69 @@ moderror(int err)
 	}
 }
 
+#define SYS_MODULES_ATTR_PATH ("/sys/module/%s/%s")
+
+static int
+get_loaded_kmod_attr(const char *attr, char *val, size_t val_len)
+{
+	FILE *attr_file;
+	char path[128];
+	char line[64];
+	int ret;
+
+	ret = snprintf(path, sizeof(path), SYS_MODULES_ATTR_PATH,
+		KNI_MODULE_NAME, attr);
+	RTE_VERIFY(ret > 0 && ret < (int)sizeof(path));
+
+	attr_file = fopen(path, "r");
+	if (attr_file == NULL) {
+		CPS_LOG(ERR, "Can't open %s: %s\n", path, strerror(errno));
+		return -1;
+	}
+
+	if (fgets(line, sizeof(line), attr_file) != NULL) {
+		size_t len;
+		strncpy(val, line, val_len - 1);
+		val[val_len - 1] = '\0';
+		len = strlen(val);
```

Comment:
We need to check that the content of buffer `line` fits in `val` AND that the buffer `line` was able to read the whole line.